### PR TITLE
Feature/product demo impl

### PR DESCRIPTION
**Pull request by Pixcell**

### CODE REVIEW

File: pkg/util/yaml.go [hunk 1]
```diff
@@ -32,14 +33,20 @@ func UnmarshalYAML(filename string, object interface{}) error {
 	return nil
 }
 
-func Print(obj interface{}) error {
+func Print(obj interface{}, filename string) error {
```
Comment (SergeRadinovich) [inline pkg/util/yaml.go]: Suggest make new function Fprint(iowriter, interface) which can be used print Print() by passing stdout to Fprint.
So Print uses Fprint(stdout,..)

File: internal/deploy/application/remote.go [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+func (exe *remoteExecutor) init(controller *config.Controller) (err error) {
+	exe.client = client.New(controller.Endpoint)
+	if err = exe.client.Login(client.LoginRequest{Email: controller.IofogUser.Email, Password: controller.IofogUser.Password}); err != nil {
+		return
+	}
+	listAgents, err := exe.client.ListAgents()
+	if err != nil {
+		return
+	}
+	exe.agentsByName = make(map[string]*client.AgentInfo)
```
Comment (SergeRadinovich) [inline internal/deploy/application/remote.go]: Could you re-use your other function which converts array to map? Assuming you can make it generic by using interface. If so, you can move it to util pkg if need

File: internal/deploy/application/remote.go [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+	for i := 0; i < len(listAgents.Agents); i++ {
+		exe.agentsByName[listAgents.Agents[i].Name] = &listAgents.Agents[i]
+	}
+
+	listCatalog, err := exe.client.GetCatalog()
+	if err != nil {
+		return
+	}
+	exe.catalogByID = make(map[int]*client.CatalogItemInfo)
```
Comment (SergeRadinovich) [inline internal/deploy/application/remote.go]: Same here as above

File: internal/deploy/application/remote.go [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+		if err != nil {
+			return
+		}
+		// CatalogItem
+		var catalogItem *client.CatalogItemInfo
+		catalogImages := []client.CatalogImage{
+			{ContainerImage: msvc.Images.X86, AgentTypeID: 1},
+			{ContainerImage: msvc.Images.ARM, AgentTypeID: 2},
+		}
+		if msvc.Images.CatalogID == 0 {
```
Comment (SergeRadinovich) [inline internal/deploy/application/remote.go]: This function gets really hard to follow at this point because of all the indentation - can we break it up into more functions?

Comment (SergeRadinovich) [reply internal/deploy/application/remote.go]: I understand the bulk of the lines is just type decl/definitions

File: internal/describe/application.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	exe.flow, err = exe.client.GetFlowByName(exe.name)
+	if err != nil {
+		return
+	}
+	msvcListResponse, err := exe.client.GetMicroservicesPerFlow(exe.flow.ID)
+	if err != nil {
+		return
+	}
+	exe.msvcs = msvcListResponse.Microservices
+	exe.msvcPerID = make(map[string]*client.MicroserviceInfo)
```
Comment (SergeRadinovich) [inline internal/describe/application.go]: another array to map conversion you might be able to use generic function for

File: internal/deploy/application/remote.go (hunk 4 above)
Comment (Pixcell) [reply internal/deploy/application/remote.go]: Done

File: internal/deploy/application/remote.go (hunk 2 above)
Comment (Pixcell) [reply internal/deploy/application/remote.go]: Without golang generics, there is too many small variations to make a generic array to map function

File: internal/deploy/application/remote.go (hunk 3 above)
Comment (Pixcell) [reply internal/deploy/application/remote.go]: Without golang generics, there is too many small variations to make a generic array to map function

File: internal/describe/application.go (hunk 5 above)
Comment (Pixcell) [reply internal/describe/application.go]: Without golang generics, there is too many small variations to make a generic array to map function

File: pkg/util/yaml.go (hunk 1 above)
Comment (Pixcell) [reply pkg/util/yaml.go]: Done
